test/wpool_SUITE.erl

```diff
@@ -168,7 +168,8 @@ stop_pool(_Config) ->
 
   {comment, []}.
 
--spec non_brutal_shutdown(config()) -> {comment, []}.
+%%-spec non_brutal_shutdown(config()) -> {comment, []}.
+-spec non_brutal_shutdown(config()) -> no_return().
```

Comment:
The code in this function seems to manifest the desire to return `{comment, []}` (see line 187 below).
If dialyzer was complaining about the spec, that's probably because of an error that happens somewhere else in the code. In other words, what was wrong was not the spec.